iOS/DuckDuckGo/TabPreviewsSource.swift

```diff
@@ -176,46 +188,17 @@ class TabPreviewsSource {
         
         return UIImage(data: data)
     }
-}
 
-class TabPreviewsCleanup {
-    
-    static let shared = TabPreviewsCleanup()
-    
-    private let lock = NSLock()
-    private var isCleaning = false
-    
-    func startCleanup(with model: TabsModel, source: TabPreviewsSource, completion: @escaping () -> Void = {}) {
-        lock.lock()
-        guard let storeDir = source.previewStoreDir, !isCleaning else {
-            lock.unlock()
-            completion()
-            return
-        }
-        isCleaning = true
-        lock.unlock()
-        
-        source.cleanupCache()
-        
-        let validIDs = Set(model.tabs.map { $0.uid })
-        
-        DispatchQueue.global(qos: .userInitiated).async {
-            if let previews = try? FileManager.default.contentsOfDirectory(at: storeDir, includingPropertiesForKeys: nil) {
-                for previewURL in previews where previewURL.lastPathComponent.hasSuffix(".png") {
-                    let previewID = previewURL.lastPathComponent.dropLast(4)
-                    
-                    if !validIDs.contains(String(previewID)) {
-                        try? FileManager.default.removeItem(at: previewURL)
-                    }
-                }
+    func removePreviewsWithIdNotIn(_ ids: Set<String>) async {
+        guard let directory = previewStoreDir else { return }
+        let contents = try? FileManager.default.contentsOfDirectory(atPath: directory.path)
```

Comment:
This loads the entire contents of the directory in memory.   What do you think about using a FileEnumerator to stream them instead?

```
guard let enumerator = FileManager.default.enumerator(
    at: directory,
    includingPropertiesForKeys: nil,
    options: [.skipsHiddenFiles]
) else { return }

for case let url as URL in enumerator {
// Remove it...
}

```

Reply:
Yeah, good idea